src/posix/platform/settings.cpp

```diff
@@ -166,10 +166,20 @@ static void swapDiscard(otInstance *aInstance, int aFd)
     VerifyOrDie(0 == unlink(swapFileName), OT_EXIT_ERROR_ERRNO);
 }
 
-void otPlatSettingsInit(otInstance *aInstance)
+void otPlatSettingsInit(otInstance *aInstance, const uint16_t *aCriticalKeys, uint16_t aCriticalKeysLength)
 {
+#if !OPENTHREAD_POSIX_CONFIG_SECURE_SETTINGS_ENABLE
+    OT_UNUSED_VARIABLE(aCriticalKeys);
+    OT_UNUSED_VARIABLE(aCriticalKeysLength);
+#endif
+
     otError error = OT_ERROR_NONE;
 
+#if OPENTHREAD_POSIX_CONFIG_SECURE_SETTINGS_ENABLE
+    sKeys       = aCriticalKeys;
+    sKeysLength = aCriticalKeysLength;
+#endif
```

Comment:
If it is possible to use the `#if #else #endif` to process the parameters, rather than using two code blocks.

Reply:
The style guide enforces the order of codes in a function:

`OT_UNUSED_VARIABLE` -> local variable definitions -> others

So we have to use 2`#if` blocks here.